libs/text-splitters/tests/unit_tests/test_text_splitters.py

```diff
@@ -2051,3 +2052,43 @@ def test_powershell_code_splitter_longer_code() -> None:
         "$csvContent | ForEach-Object {\n    $_.ProcessName\n}",
         "# End of script",
     ]
+
+
+@pytest.mark.requires("nltk")
+def test_nltk_text_splitter_args() -> None:
+    """Test invalid arguments for NLTKTextSplitter."""
+    import nltk  # type: ignore[import-not-found]
+
+    nltk.download("punkt_tab")
+    with pytest.raises(ValueError):
+        NLTKTextSplitter(
+            chunk_size=80,
+            chunk_overlap=0,
+            separator="\n\n",
+            use_span_tokenize=True,
+        )
+
+
+@pytest.mark.requires("nltk")
+def test_nltk_text_splitter_with_add_start_index() -> None:
+    import nltk  # type: ignore[import-not-found]
+
+    nltk.download("punkt_tab")
```

Comment:
if this is needed, could you update docs and add an error message telling people to run this?

Reply:
Yes, this is needed. For example here there is the CI that failed without that download: https://github.com/langchain-ai/langchain/actions/runs/11609762759/job/32327689651

If you agree, we can leave it as it is in `libs/text-splitters/tests/unit_tests/test_text_splitters.py` and just add a message in `NLTKTextSplitter`'s docstring in `libs/text-splitters/langchain_text_splitters/nltk.py`

Reply:
got it. If there's a way to mock this as well, that would be great. This made me realize we weren't running these unit tests without network access in the same way we do in other packages. Fixed here: #28732

How would be best to test this without network access?

Reply:
(just merged in master, which should theoretically fail the test with the proper pytest socket blocking)

Reply:
added some commits to make this a network-allowed integration test instead of a unit test. won't run on prs, but that's ok